src/modules/cmake.rs

```diff
@@ -0,0 +1,78 @@
+use super::{Context, Module, RootModuleConfig};
+
+use crate::configs::cmake::CMakeConfig;
+use crate::formatter::StringFormatter;
+use crate::utils;
+
+/// Creates a module with the current CMake version
+///
+/// Will display the CMake version if any of the following criteria are met:
+///     - The current directory contains a `CMakeLists.txt` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_cmake_project = context
+        .try_begin_scan()?
+        .set_files(&["CMakeLists.txt"])
+        .is_match();
+
+    if !is_cmake_project {
+        return None;
+    }
+
+    let cmake_version = utils::exec_cmd("cmake", &["--version"])?.stdout;
+    let module_version = format_cmake_version(&cmake_version)?;
+
+    let mut module = context.new_module("cmake");
+    let config: CMakeConfig = CMakeConfig::try_load(module.config);
+    let formatter = if let Ok(formatter) = StringFormatter::new(config.format) {
+        formatter.map(|variable| match variable {
+            "version" => Some(module_version.clone()),
+            _ => None,
+        })
+    } else {
+        log::warn!("Error parsing format string in `cmake.format`");
+        return None;
+    };
```

Comment:
Check https://github.com/starship/starship/pull/1063/files as an example. The upstream branch `feat/format-string` has updated the API of `StringFormatter`.